plugins/inputs/ldap/ldap.go

```diff
@@ -131,31 +151,39 @@ func (l *LDAP) Gather(acc telegraf.Accumulator) error {
 func (l *LDAP) connect() (*ldap.Conn, error) {
 	var conn *ldap.Conn
 	switch l.mode {
-	case "ldap":
+	case "ldap", "ldapi":
 		var err error
-		conn, err = ldap.DialURL("ldap://" + l.Server)
+		conn, err = ldap.DialURL(l.Server)
 		if err != nil {
 			return nil, err
 		}
-	case "ldaps":
-		var err error
-		conn, err = ldap.DialURL("ldaps://"+l.Server, ldap.DialWithTLSConfig(l.tlsCfg))
-		if err != nil {
-			return nil, err
+		if *l.ClientConfig.Enable {
+			if err := conn.StartTLS(l.tlsCfg); err != nil {
+				return nil, err
+			}
 		}
-	case "starttls":
+	case "ldaps":
 		var err error
-		conn, err = ldap.DialURL("ldap://" + l.Server)
+		conn, err = ldap.DialURL(l.Server, ldap.DialWithTLSConfig(l.tlsCfg))
 		if err != nil {
 			return nil, err
 		}
-		if err := conn.StartTLS(l.tlsCfg); err != nil {
-			return nil, err
-		}
 	default:
 		return nil, fmt.Errorf("invalid tls_mode: %s", l.mode)
 	}
```

Comment:
See above, keep the URL and pass this on to the dial call. If we really need to do something special for `ldapi` then add a `case`...